src/towncrier/create.py

```diff
@@ -59,11 +61,32 @@ def __main(directory, config, filename):
     if os.path.exists(segment_file):
         raise click.ClickException("{} already exists".format(segment_file))
 
+    content = _get_news_content(edit)
+
+    if content is None:
+        click.echo("Abort creating news fragment.")
+        ctx.exit(1)
```

Comment:
Do you know the details of the various ways to exit a click program and which is best?  (and why `:]`)  I wasn't even aware that `ctx.exit()` existed.

Reply:
`ctx.exit` means exit the program with code.
You can always use the python `return` directive if you want, but it means that the exit code would be 0.

Here, I chose to exit with code 1 because if the user did not write content, it means the operation has "failed", and the exit code should reflect that.

As for the use of `ctx.exit`, *click* has this **real bad habit** of wrapping built-in python methods with *click* functions (for example, wrapping the `print` method in `click.echo`). I don't like it at all, but I take it as a conversion when working with *click* (this is related to our mocking discussion). In our case, we could simply use `sys.exit` instead and it would do the same.

To sum up, `ctx.exit` is the best choice, considering that it is a convention.

Reply:
I agree a non-zero exit code is sensible, my only question was how to do it.  I asked about this in the Click discord and the only answer I got was `sys.exit()`, but I think they hadn't heard of `ctx.exit()` either.  `:]`  Anyways, `click.echo()` does do more than print especially when dealing across Python versions and platforms.  There's lots of room for major hassle when trying to handle that at the lower levels.